### DISCUSSION TITLE
Trying to mock axios.get with Promise.Reject to trigger onError in useQuery

### DISCUSSION THREAD

**Discussion by emil45** [TanStack/query#2196] [Q&A]
Hello,

I'm trying to trigger `onError` in `useQuery` by mocking the `axios.get` with `Promise.reject`, but `onError` never triggered.
Trying to understand if there is any bug or something wrong with the code?

My `useQuery` function that is called in the test:

```
function useGetSomething() {
	return useQuery('getSomething', AxiosService.getSomething, {
		onError: (error) => LoggerService.error('Failed to fetch the something', { error })
	});
}
```

```
const mockedGetSomething = jest.spyOn(AxiosService, 'get').mockImplementation(() => {
    return Promise.reject('error');
});
```

Is there any problem to mock the `axios.get` ? or it's a problem with `react-query` ?

**Comment by frendyguo-zz** [comment]
Based on what you're trying to do,
shouldn't `jest.spyOn(AxiosService, 'get')` be `jest.spyOn(AxiosService, 'getSomething')`?
Because that's what you're trying to fake right?

**Comment by TkDodo** [comment]
have you turned off `retries` in the `QueryClientProvider` ? Otherwise, react-query will retry failures 3 times and it will likely timeout.

here is my take on testing react-query custom hooks and components: https://tkdodo.eu/blog/testing-react-query

**Reply by AndreSilva1993** [reply]
Faced the same one year and a half later, still holds up. Thanks man!

**Reply by seanpmaxwell** [reply]
Me too!
